prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -282,9 +289,56 @@ public synchronized APIResult update(HttpServletRequest request, String type, St
             throw FalconWebException.newException(e, Response.Status.BAD_REQUEST);
         } finally {
             ConfigurationStore.get().cleanupUpdateInit();
+            releaseUpdateEntityLocks(entityName, tokenList);
```

Comment:
How are we handling the scenario when obtainUpdateEntityLocks() throws exception after acquiring a partial set of locks? In that case, tokenList will still be null, thus blocking all operations that need to acquire any lock in that partial subset.